k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java

```diff
@@ -0,0 +1,236 @@
+package com.fsck.k9.ui;
+
+
+import android.content.AsyncQueryHandler;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.graphics.Canvas;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.ContactsContract.CommonDataKinds.Email;
+import android.provider.ContactsContract.Contacts;
+import android.provider.ContactsContract.Intents;
+import android.provider.ContactsContract.QuickContact;
+import android.provider.ContactsContract.RawContacts;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityNodeInfo;
+import android.widget.ImageView;
+
+
+/**
+ * ContactBadge replaces the android ContactBadge for custom drawing.
+ * <p>
+ * Based on QuickContactBadge:
+ * https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/QuickContactBadge.java
+ */
+public class ContactBadge extends ImageView implements OnClickListener {
+    static final private int TOKEN_EMAIL_LOOKUP = 0;
+    static final private int TOKEN_EMAIL_LOOKUP_AND_TRIGGER = 1;
+
+    static final private String EXTRA_URI_CONTENT = "uri_content";
+
+    static final String[] EMAIL_LOOKUP_PROJECTION = new String[] {
+            RawContacts.CONTACT_ID,
+            Contacts.LOOKUP_KEY,
+    };
+    static final int EMAIL_ID_COLUMN_INDEX = 0;
+    static final int EMAIL_LOOKUP_STRING_COLUMN_INDEX = 1;
+
+
+    private Uri contactUri;
+    private String contactEmail;
+    private QueryHandler queryHandler;
+    private Bundle extras = null;
+
+
+    public ContactBadge(Context context) {
+        this(context, null);
+    }
+
+    public ContactBadge(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public ContactBadge(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        queryHandler = new QueryHandler(context.getContentResolver());
+        setOnClickListener(this);
+    }
+
+    @Override
+    protected void drawableStateChanged() {
+        super.drawableStateChanged();
+    }
+
+    /**
+     * This call has no effect anymore, as there is only one QuickContact mode
+     */
+    @SuppressWarnings("unused")
+    public void setMode(int size) {
```

Comment:
This method doesn't seem to be used from any of our code. And it doesn't override a method of the parent class. Why is it here?